driver-core/src/main/com/mongodb/internal/connection/BaseCluster.java

```diff
@@ -376,24 +383,31 @@ private MongoIncompatibleDriverException createIncompatibleException(final Clust
         return new MongoIncompatibleDriverException(message, curDescription);
     }
 
-    private MongoTimeoutException createTimeoutException(final ServerSelector serverSelector, final ClusterDescription curDescription) {
-        return new MongoTimeoutException(format("Timed out after %d ms while waiting for a server that matches %s. "
-                                                + "Client view of cluster state is %s",
-                                                settings.getServerSelectionTimeout(MILLISECONDS), serverSelector,
-                                                curDescription.getShortDescription()));
+    private MongoException createAndLogTimeoutException(
+            final OperationContext operationContext,
+            final ServerSelector serverSelector,
+            final ClusterDescription clusterDescription) {
+        MongoTimeoutException exception = new MongoTimeoutException("Timed out while waiting for a suitable server");
```

Comment:
The existing message that we use for this exception has proved incredibly useful for support over the years.  I don't think we should change it.  Is there something in the spec that requires this change?

I'd be fine if the exception message more closely matches what we now log for failures.

Reply:
I think I see why you did this.  Is it because you log the message from this exception and don't want to repeat anything?  If so, I think we should just log the message here as a constant and leave the message as is.

Reply:
The spec [requires the following regarding the `failure` key](https://github.com/mongodb/specifications/blob/master/source/server-selection/server-selection.rst#server-selection-failed-message):

> Drivers MUST take care to not include any information in this field that is already included in the log message; e.g. the topology description should not be duplicated within this field.

Given that both the selector and the topology description must be included in the log message according to the logging spec, the only piece of data we are allowed to still have in the `MongoTimeoutException` message of those we previously had, is the amount of time after which we timed out. This was already discussed, and everyone was happy that we don't need to track it in CSOT because the spec does not require us to log it.

> so, I think we should just log the message here as a constant and leave the message as is.

By "as is", do you mean to keep the selector and the topology description in the `MongoTimeoutException` message, or do you mean to also preserve the amount of time after which we timed out? The latter invalidates what was discussed previously in the CSOT context.